numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -932,7 +970,9 @@ def compile(self, sig):
             kernel = _Kernel(self.py_func, argtypes, **self.targetoptions)
             # We call bind to force codegen, so that there is a cubin to cache
             kernel.bind()
-            self._cache.save_overload(sig, kernel)
+            # We only cache kernels that has no external linkage.
+            if len(kernel._codelibrary._linking_files) == 0:
+                self._cache.save_overload(sig, kernel)
```

Comment:
This change should be undone now, because we're not linking the NRT for any non-NRT tests, and we're not caching anything that does use NRT in tests (we can defer support for caching functions that use NRT to a later PR).
```suggestion
            self._cache.save_overload(sig, kernel)
```